### PR TITLE
move linters to pre-commit

### PR DESCRIPTION
**Pull request by graingert**
It's recommended to run pyflakes via flake8, and to run flake8 via pre-commit. So I configured them here.

I also configured prettier, and some other pre-commit hooks at the same time.

The advantage of running flake8 via pre-commit is that it also checks files that contain python code, but don't end in `.py`, So I fixed the issues there.

## Contributor Checklist:

* [ ] I have updated the release notes at `docs/source/NEWS.rst`
* [x] I have updated the automated tests.
* [x] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -74,3 +74,53 @@ universal = 1
...
+
+
+[flake8]
+disable-noqa = True
+max-line-length = 368
+extend-ignore =
+
+   E203,  # whitespace before : is not PEP8 compliant (& conflicts with black)
+
+   E101,  # indentation contains mixed spaces and tabs
```
Comment (graingert) [inline setup.cfg:86]: anything starting `Exxx` or `Wxxx` is a pycodestyle lint failure. `Fxxx` is for pyflakes

Comment (graingert) [reply setup.cfg:86]: almost all of these get fixed by black: https://github.com/twisted/ldaptor/pull/196/files#diff-380c6a8ebbbce17d55d50ef17d3cf906R85

**Review by adiroiban — COMMENTED** [review]
will continue the review later

File: setup.cfg (hunk 1 above)
Comment (graingert) [reply setup.cfg:86]: @adiroiban I'm going to do black in a follow-up PR as it's a huge number of changes and to review it I'm going to ask you to run "pipx run pre-commit run --all-files" until it passes and then get you to verify the diff

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**